src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.c

```diff
@@ -332,8 +332,15 @@ static void pf_update_locked(grpc_exec_ctx *exec_ctx, grpc_lb_policy *policy,
     return;
   }
   if (GRPC_TRACER_ON(grpc_lb_pick_first_trace)) {
-    gpr_log(GPR_INFO, "Pick First %p received update with %lu addresses",
+    gpr_log(GPR_INFO, "Pick First %p received update with %lu addresses.",
             (void *)p, (unsigned long)num_addrs);
+    for (size_t i = 0; i < addresses->num_addresses; ++i) {
+      if (addresses->addresses[i].is_balancer) continue;
+      char *address_uri =
+          grpc_sockaddr_to_uri(&addresses->addresses[i].address);
+      gpr_log(GPR_DEBUG, "\t-> Update address: %s", address_uri);
```

Comment:
Tabs don't always show up well in logs.  I suggest just using a couple of spaces here.